### PR TITLE
EZEE-3303 - search engine comparison

### PR DESCRIPTION
**Pull request by dabrt**
| Question | Answer |
| ------------- | --- |
| JIRA Ticket   | EZEE-3303 |
| Versions      | master |

Add a table with search engine comparison to developer's guide.

### CODE REVIEW

File: docs/guide/search/search.md [hunk 1]
```diff
@@ -2,10 +2,40 @@
...
+sorting on scoring by default, for location search available as of [v1.3][3] with Solr 6.x), 
+and as of [Solr Search Engine v1.5](https://github.com/ezsystems/ezplatform-solr-search-engine/releases/tag/v1.5.0) supports advanced 
+full text capabilities such as `word "phrase" (group) +mandatory -prohibited AND && OR || NOT !`._
+
+_*** As of Solr 7, index-time boosting is no longer available. Query time boosting is planned to be introduced instead in future 2.x and 3.x release.
+See [Major changes in Solr 7](https://lucene.apache.org/solr/guide/7_2/major-changes-in-solr-7.html) for more information._
+
+_**** Elasticsearch offers query time boosting instead._
+
+## Install
```
Comment (dabrt) [inline docs/guide/search/search.md:38]:
```suggestion
```

**Review by adamwojs — APPROVED** [review]

File: docs/guide/search/search.md [hunk 2]
```diff
@@ -2,10 +2,39 @@
...
+Currently, the following search engines exist in their own eZ Platform Bundles:
 
 1.  [Legacy search engine](search_engines.md#legacy-search-engine-bundle), a database-powered search engine for basic needs.
-1.  [Solr](solr.md), an integration providing better overall performance, much better scalability and support for more advanced search capabilities **(recommended)**
+1.  [Solr](solr.md), an integration providing better overall performance, much better scalability and support for more advanced search capabilities.
+1.  [Elasticsearch](elastic.md), available for eZ Enterprise customers, a document-oriented engine providing even better performance and scalability.
+
+### Feature comparison
+
+| **Feature** | **Elasticsearch** | **Apache Solr** | **Legacy Search Engine** _**(SQL)**_ |
```
Comment (DominikaK) [inline docs/guide/search/search.md:13]:
```suggestion
| Feature | Elasticsearch | Apache Solr | Legacy Search Engine (SQL) |
```

File: docs/guide/search/search.md [hunk 3]
```diff
@@ -2,10 +2,39 @@
...
+| --- | --- | --- | --- |
+| Filtering | Yes | Yes | Yes, but limited\* |
```
Comment (DominikaK) [inline docs/guide/search/search.md:15]:
```suggestion
| Filtering | Yes | Yes | Yes, limited\* |
```

File: docs/guide/search/search.md [hunk 4]
```diff
@@ -2,10 +2,39 @@
...
+| Query _(filter with scoring)_ | Yes | Yes | Only filters, no scoring |
```
Comment (DominikaK) [inline docs/guide/search/search.md:16]:
```suggestion
| Query (filter with scoring) | Yes | Yes | Only filters, no scoring |
```

File: docs/guide/search/search.md [hunk 5]
```diff
@@ -2,10 +2,39 @@
...
+| Fulltext | Yes | Yes, incl. advanced features | Partly\*\* |
```
Comment (DominikaK) [inline docs/guide/search/search.md:17]:
```suggestion
| Fulltext | Yes | Yes, incl. advanced features | Partial support \*\* |
```

File: docs/guide/search/search.md [hunk 6]
```diff
@@ -2,10 +2,39 @@
...
+| Faceting | Partly (deprecated) | [Partly as of v1.4](https://doc.ezplatform.com/en/latest/api/public_php_api_browsing/#performing-a-faceted-search) | No |
```
Comment (DominikaK) [inline docs/guide/search/search.md:18]:
```suggestion
| Faceting (deprecated)  | Partial support | Partial support | No |
```
The oldest supported version uses the solr bundle > v1.4, so this is irrelevant.
Also remember to remove the "deprecated" note for faceting when merging to other versions (it is only deprecated in v3.2).

File: docs/guide/search/search.md [hunk 7]
```diff
@@ -2,10 +2,39 @@
...
+| Spellchecking | Planned _(TBD when)_ | Planned _(TBD when)_ | No |
+| Highlighting | Planned _(TBD when)_ | Planned _(TBD when)_ | No |
```
Comment (DominikaK) [inline docs/guide/search/search.md:20]:
```suggestion
| Spellchecking | No | No | No |
| Highlighting | No | No | No |
```
Let's deal with what is in the product now, not in the future. Or you could just remove those two lines altogether

File: docs/guide/search/search.md [hunk 8]
```diff
@@ -2,10 +2,39 @@
...
+| Index time boosting | ?\*\*\*\* | [Yes, in \&gt;= v1.4,\&lt; v2.0](https://github.com/ezsystems/ezplatform-solr-search-engine/releases/tag/v1.4.0)\*\*\* | No |
```
Comment (DominikaK) [inline docs/guide/search/search.md:21]:
```suggestion
| Index time boosting | \*\*\*\* | Yes | No |
```
AFAIK this means it is supported in v3, but not in earlier versions. Please indicate it when merging to relevant branches.

File: docs/guide/search/search.md [hunk 9]
```diff
@@ -2,10 +2,39 @@
...
+| Aggregations | Yes | Yes | No |
+
+
+_* Usage of Criterion and SortClause for Fields does not perform well on medium to larger
```
Comment (DominikaK) [inline docs/guide/search/search.md:25]:
```suggestion
* Usage of Criteria and Sort Clauses for Fields does not perform well on medium to larger 
```

File: docs/guide/search/search.md [hunk 10]
```diff
@@ -2,10 +2,39 @@
...
+| Fulltext | Yes | Yes, incl. advanced features | Partly\*\* |
+| Faceting | Partly (deprecated) | [Partly as of v1.4](https://doc.ezplatform.com/en/latest/api/public_php_api_browsing/#performing-a-faceted-search) | No |
+| Spellchecking | Planned _(TBD when)_ | Planned _(TBD when)_ | No |
+| Highlighting | Planned _(TBD when)_ | Planned _(TBD when)_ | No |
+| Index time boosting | ?\*\*\*\* | [Yes, in \&gt;= v1.4,\&lt; v2.0](https://github.com/ezsystems/ezplatform-solr-search-engine/releases/tag/v1.4.0)\*\*\* | No |
+| Aggregations | Yes | Yes | No |
+
+
+_* Usage of Criterion and SortClause for Fields does not perform well on medium to larger 
+amount of data with Legacy Search Engine (SQL), use Solr for this._
```
Comment (DominikaK) [inline docs/guide/search/search.md:26]:
```suggestion
amounts of data with Legacy Search Engine (SQL), use Solr for this.
```

File: docs/guide/search/search.md [hunk 11]
```diff
@@ -2,10 +2,39 @@
...
+
+_** Does not include full set of full text features includes: Scoring/Ranking (Solr performs 
+sorting on scoring by default, for location search available as of [v1.3][3] with Solr 6.x), 
+and as of [Solr Search Engine v1.5](https://github.com/ezsystems/ezplatform-solr-search-engine/releases/tag/v1.5.0) supports advanced 
+full text capabilities such as `word "phrase" (group) +mandatory -prohibited AND && OR || NOT !`._
```
Comment (DominikaK) [inline docs/guide/search/search.md:31]:
```suggestion
** Does not cover all full text features including: scoring/ranking (Solr performs sorting on scoring by default)
```
Please double check, but I think all the warnings in this point referred to versions of the solr bundle that are not used in any supported version of Platform (v1.13 uses solr bundle v1.5)

File: docs/guide/search/search.md [hunk 12]
```diff
@@ -2,10 +2,39 @@
...
+
+_*** As of Solr 7, index-time boosting is no longer available. Query time boosting is planned to be introduced instead in future 2.x and 3.x release.
```
Comment (DominikaK) [inline docs/guide/search/search.md:33]:
```suggestion
*** As of Solr 7, index-time boosting is no longer available.
```

File: docs/guide/search/search.md [hunk 13]
```diff
@@ -2,10 +2,39 @@
...
+See [Major changes in Solr 7](https://lucene.apache.org/solr/guide/7_2/major-changes-in-solr-7.html) for more information._
```
Comment (DominikaK) [inline docs/guide/search/search.md:34]:
```suggestion
See [Major changes in Solr 7](https://lucene.apache.org/solr/guide/7_2/major-changes-in-solr-7.html) for more information.
```

File: docs/guide/search/search.md [hunk 14]
```diff
@@ -2,10 +2,39 @@
...
+
+_**** Elasticsearch offers query time boosting instead._
```
Comment (DominikaK) [inline docs/guide/search/search.md:36]:
```suggestion
**** Elasticsearch offers query time boosting instead.
```

**Review by DominikaK — CHANGES_REQUESTED** [review]

File: docs/guide/search/search.md [hunk 15]
```diff
@@ -2,10 +2,27 @@
...
+1.  [Solr](solr.md), an integration providing better overall performance, much better scalability and support for more advanced search capabilities.
+1.  [Elasticsearch](elastic.md), available for eZ Enterprise customers, a document-oriented engine providing even better performance and scalability.
+
+### Feature comparison
+
+| Feature | Elasticsearch | Apache Solr | Legacy Search Engine (SQL) |
+| --- | --- | --- | --- |
+| Filtering | Yes | Yes | Yes, limited\* |
+| Query (filter with scoring) | Yes | Yes | Only filters, no scoring |
+| Fulltext | Yes | Yes, incl. advanced features | Partial support |
```
Comment (DominikaK) [inline docs/guide/search/search.md:17]:
```suggestion
| Fulltext search | Yes | Yes, incl. advanced features | Partial support |
```

**Review by DominikaK — APPROVED** [review]

**Pull request merged**